src/server/transaction.cc

```diff
@@ -1126,9 +1096,20 @@ KeyLockArgs Transaction::GetLockArgs(ShardId sid) const {
   return res;
 }
 
-bool Transaction::IsArmedInShard(ShardId sid) const {
-  // Barrier has acquire semantics
-  return run_barrier_.Active() && (shard_data_[SidToId(sid)].local_mask & ARMED);
+uint16_t Transaction::DisarmInShard(ShardId sid) {
+  auto& sd = shard_data_[SidToId(sid)];
+  // NOTE: Maybe compare_exchange is worth it to avoid redundant writes
```

Comment:
my understanding that LOCK XCHG is considered a "heavy" atomic op and is much slower than other atomics.

see community.intel.com/t5/Intel-Moderncode-for-Parallel/lock-xchg-performance/m-p/917111#M4845
for example

Reply:
I assume you mean CMPXCHG? It's not that much slower https://stackoverflow.com/questions/5339769/relative-performance-of-swap-vs-compare-and-swap-locks-on-x86 and it saves a line invalidation. On the other head the onwing thread only writes to is_armed, so it doesn't care about ownership mostly